src/BotPlutusInterface/CoinSelection.hs

```diff
@@ -0,0 +1,479 @@
+
+                   [("scriptHash1", "TokenName1"), ("", "")]
+    output-vector: [1,                             6] -- Just Adding output1 + output2 + Fee
+
+    utxo1-vector: [10,                             10]
+    utxo2-vector: [0,                              5]
+    utxo3-vector: [3,                              6]
+
+  Now, as stated above our goal is to get as close to output-vector as possible,
+  but we also need to satisfiy the following condition:
+
+    1. Each column of the resultant vector must be greater than or equal to the corresponding
+       column of the output vector.
+
+       In this case this means that, the resultant vector [x , y] should be such that
+       x >= 1 and y >= 6.
+
+    2. The input vector must not be the zero vector.
+
+  Currently we don't have any inputs to the Tx, hence the input vector will be zero vector:
+
+    input-vector: [0, 0]
+
+  Now, we can start searching for utxos that can statisfy our goal and mission:
+
+    step 1. Add input vector to all the utxo vectors. In this case these vectors will be
+            utxo1-vector, utxo2-vector, utxo3-vector.
+
+            Result: [10, 10], [0, 5], [3, 6] (adding zero vector does not change the vector).
+
+    step 2. calculate distances of previously added vectors with output vector.
+
+            Result: [ 9.84 -- l2norm(output-vector, utxo1-vector)
+                    , 1.41 -- l2norm(output-vector, utxo2-vector)
+                    , 2.00 -- l2norm(output-vector, utxo3-vector)
+                    ]
+
+            As, we can see the distance between utxo1-vector and output-vector is very large
+            which is to be expected as utxo1-vector contains lots of "value" of different assetclass.
+
+    step 3. sort the distances, and select the utxos with least distances
+            until all the conditions are statisfied.
+
+            Result: [ 1.41
+                    , 2.00
+                    , 9.84
+                    ]
+
+            Since, utxo2-vector has the least distance we will select that utxo.
+
+            But, selecting utxo2-vector alone doesn't statisfy all our conditions,
+            hence we will have to continue selecting. After utxo2-vector, the vector with
+            least distance is utxo3-vector, hence we will select that vector.
+
+            Now, the input vector is: [3, 11] (which is just addition of utxo2-vector & utxo3-vector).
+
+            Since, this input vector statisfy all our conditions, we will select utxo2 & utxo3 as inputs
+            to our Tx.
+
+-}
+
+-- 'searchStrategy' represents the possible search strategy.
+data SearchStrategy
+  = -- | This is a greedy search that searches for nearest utxo using l2norm.
+    Greedy
+  | -- | This is like greedy search, but here there's
+    -- additonal goal that the change utxo should be equal to the output utxo.
+    GreedyApprox
+  deriving stock (Eq, Show)
+
+instance Default SearchStrategy where
```

Comment:
I mentioned this in the balancing PR, but I'd personally recommend against `data-default` in general

Reply:
fixed